File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    answer = True
    hash_map = {}
    for nums in phone_book:
        hash_map[nums] = 1

    
    for nums in phone_book:
        front = ""
        for num in nums:
            front += num
            if front in hash_map and front != nums:
                answer = False
                break
    return answer

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        int n = phone_book.length;
     
        Arrays.sort(phone_book);
        
        for (int i = 0; i < n - 1; i++) {
            if (phone_book[i + 1].startsWith(phone_book[i])) {
                return false;
            }
        }
        
        return answer;
    }
}